packages/libs/eda/src/lib/map/analysis/mapTypes/plugins/BubbleMarkerMapType.tsx

```diff
@@ -0,0 +1,673 @@
+};
+
+const getBoundsAndPosition = (
+  minLat: number,
+  minLon: number,
+  maxLat: number,
+  maxLon: number,
+  avgLat: number,
+  avgLon: number
+) => ({
+  bounds: {
+    southWest: { lat: minLat, lng: minLon },
+    northEast: { lat: maxLat, lng: maxLon },
+  },
+  position: { lat: avgLat, lng: avgLon },
+});
+
+interface OverlayConfigProps {
+  selectedVariable?: VariableDescriptor;
+  studyId: string;
+  filters?: Filter[];
+  aggregator?: BubbleMarkerConfiguration['aggregator'];
+  numeratorValues?: BubbleMarkerConfiguration['numeratorValues'];
+  denominatorValues?: BubbleMarkerConfiguration['denominatorValues'];
+}
+
+function useOverlayConfig(props: OverlayConfigProps) {
+  const {
+    studyId,
+    filters = [],
+    aggregator,
+    numeratorValues,
+    denominatorValues,
+    selectedVariable,
+  } = props;
+  const findEntityAndVariable = useFindEntityAndVariable();
+  const entityAndVariable = findEntityAndVariable(selectedVariable);
+
+  if (entityAndVariable == null)
+    throw new Error(
+      'Invalid selected variable: ' + JSON.stringify(selectedVariable)
+    );
+  const { entity: overlayEntity, variable: overlayVariable } =
+    entityAndVariable;
+  // If the variable or filters have changed on the active marker config
+  // get the default overlay config.
+  return useMemo(() => {
+    return getDefaultBubbleOverlayConfig({
+      studyId,
+      filters,
+      overlayVariable,
+      overlayEntity,
+      aggregator,
+      numeratorValues,
+      denominatorValues,
+    });
+  }, [
+    studyId,
+    filters,
+    overlayVariable,
+    overlayEntity,
+    aggregator,
+    numeratorValues,
+    denominatorValues,
+  ]);
+}
+
+interface DataProps {
+  boundsZoomLevel?: BoundsViewport;
+  configuration: BubbleMarkerConfiguration;
+  geoConfigs: GeoConfig[];
+  studyId: string;
+  filters?: Filter[];
+}
+
+function useLegendData(props: DataProps) {
+  const { boundsZoomLevel, configuration, geoConfigs, studyId, filters } =
+    props;
+
+  const studyEntities = useStudyEntities();
+
+  const dataClient = useDataClient();
+
+  const { selectedVariable, numeratorValues, denominatorValues, aggregator } =
+    configuration as BubbleMarkerConfiguration;
+
+  const { outputEntity, geoAggregateVariables } = useCommonData(
+    selectedVariable,
+    geoConfigs,
+    studyEntities,
+    boundsZoomLevel
+  );
+
+  const outputEntityId = outputEntity?.id;
+
+  const overlayConfig = useOverlayConfig({
+    studyId,
+    filters,
+    selectedVariable,
+    aggregator,
+    numeratorValues,
+    denominatorValues,
+  });
+
+  const disabled =
+    numeratorValues?.length === 0 ||
+    denominatorValues?.length === 0 ||
+    !validateProportionValues(numeratorValues, denominatorValues);
+
+  const legendRequestParams: StandaloneMapBubblesLegendRequestParams = {
+    studyId,
+    filters: filters || [],
+    config: {
+      outputEntityId,
+      colorLegendConfig: {
+        geoAggregateVariable: geoAggregateVariables.at(-1)!,
```

Comment:
How do I get my emacs IDE to recognise `at()` 
https://blog.saeloun.com/2022/02/24/ecmascript2022-adds-add-method/

Reply:
First, make sure your packages are up to date. I'd have to learn more about your configuration to provide additional assistance.